core/src/main/java/io/questdb/griffin/engine/functions/str/LPadFunctionFactory.java

```diff
@@ -0,0 +1,103 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.str;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.sql.Function;
+import io.questdb.cairo.sql.Record;
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.StrFunction;
+import io.questdb.griffin.engine.functions.BinaryFunction;
+import io.questdb.std.IntList;
+import io.questdb.std.Numbers;
+import io.questdb.std.Chars;
+import io.questdb.std.ObjList;
+import io.questdb.std.str.StringSink;
+import org.jetbrains.annotations.Nullable;
+
+public class LPadFunctionFactory implements FunctionFactory {
+    @Override
+    public String getSignature() {
+        return "lpad(SI)";
+    }
+
+    @Override
+    public Function newInstance(int position, ObjList<Function> args, IntList argPositions, CairoConfiguration configuration, SqlExecutionContext sqlExecutionContext) throws SqlException {
+        return new LPadFunc(args.getQuick(0), args.getQuick(1));
+    }
+
+    public static class LPadFunc extends StrFunction implements BinaryFunction {
+
+        private final Function strFunc;
+        private final Function countFunc;
+
+        private final StringSink sinkA = new StringSink();
+        private final StringSink sinkB = new StringSink();
+
+        public LPadFunc(Function strFunc, Function countFunc) {
+            this.strFunc = strFunc;
+            this.countFunc = countFunc;
+        }
+
+        @Override
+        public Function getLeft() {
+            return strFunc;
+        }
+
+        @Override
+        public Function getRight() {
+            return countFunc;
+        }
+
+        @Override
+        public CharSequence getStr(final Record rec) {
+            return lPad(getLeft().getStr(rec),getRight().getInt(rec), sinkA);
```

Comment:
Please format the code properly here and in all other classes:
```suggestion
            return lPad(getLeft().getStr(rec), getRight().getInt(rec), sinkA);
```

nit: `getLeft()` and `getRight()` could be replaced with `strFunc` and `countFunc`.